src/Documenter.jl

```diff
@@ -395,8 +395,8 @@ function deploydocs(;
 
     repo_ok = occursin(travis_repo_slug, repo)
     pr_ok = travis_pull_request == "false"
-    tag_ok = !isempty(travis_tag) && occursin(Base.VERSION_REGEX, travis_tag)
-    branch_ok = isempty(travis_tag) && travis_branch == devbranch
+    tag_ok = isempty(travis_tag) || occursin(Base.VERSION_REGEX, travis_tag)
+    branch_ok = !isempty(travis_tag) || travis_branch == devbranch
```

Comment:
And this should be true if (i) the tag is not empty (and thus checked for validity on the line above) or (ii) the branches match.